### PR TITLE
[core] Clarify the purpose of method findOverlappedSnapshots and move…

### PR DESCRIPTION
**Pull request by LinMingQiang**
… to snapshotManager.

### Purpose

Clarify the purpose of method `findOverlappedSnapshots` is to find the overlapping snapshots between snapshots that still referenced by tags/branches and range of [beginInclusive, endExclusive), not only tagSnapshots.

Linked issue: close #xxx

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/operation/FileDeletionBase.java [hunk 1]
```diff
@@ -78,10 +78,11 @@ public abstract class FileDeletionBase<T extends Snapshot> {
     protected boolean changelogDecoupled;
 
     /** Used to record which tag is cached. */
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/operation/FileDeletionBase.java:80]: modify comment too

File: paimon-core/src/main/java/org/apache/paimon/operation/FileDeletionBase.java [hunk 2]
```diff
@@ -78,10 +78,11 @@ public abstract class FileDeletionBase<T extends Snapshot> {
...
-    private long cachedTag = 0;
+    private long cachedSnapshotId = 0;
 
     /** Used to cache data files used by current tag. */
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/operation/FileDeletionBase.java:83]: ditto

**Comment by LinMingQiang** [comment]
done

**Comment by LinMingQiang** [comment]
These unstable tests can be ignored, it was successful in my workflow.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**